src/ViewLayer/RaceModeDisplay/RaceModeDashboardView.cpp

```diff
@@ -187,15 +189,89 @@ void RaceModeDashboardView::initalizeFaultAnimation()
 
 void RaceModeDashboardView::runFaultAnimation()
 {
-    int numberOfActiveFaults = batteryFaultsList_.numberOfActiveFaults() +
-                               motorOneFaultsList_.numberOfActiveFaults() +
-                               motorZeroFaultsList_.numberOfActiveFaults();
+    QVector<QString> batteryFaults = batteryFaultsList_.activeFaultLabels();
+    QVector<QString> motorZeroFaults = motorZeroFaultsList_.activeFaultLabels();
+    QVector<QString> motorOneFaults = motorOneFaultsList_.activeFaultLabels();
 
-    if (numberOfActiveFaults > 0 && faultAnimation_->state() != QAbstractAnimation::Running)
+    if (faultAnimation_->state() != QAbstractAnimation::Running && triggerFaultAnimation(batteryFaults, motorZeroFaults, motorOneFaults))
     {
         faultAnimation_->setDirection(QAbstractAnimation::Forward);
         faultAnimation_->start();
     }
+
+    prevBatteryFaults_ = batteryFaults;
+    prevMotorZeroFaults_ = motorZeroFaults;
+    prevMotorOneFaults_ = motorOneFaults;
+}
+
+bool RaceModeDashboardView::triggerFaultAnimation(QVector<QString>& batteryFaults, QVector<QString>& motorZeroFaults, QVector<QString>& motorOneFaults)
+{
+
+//Will remove this comment right before merging
+//    QString str;
+//    for (int i = 0; i < batteryFaults.size(); ++i)
+//    {
+//        if (i > 0)
+//            str += "|||||";
+//        str += batteryFaults[i];
+//    }
+//    qDebug() << str << "\n";
+
+//    str = "";
+//    for (int i = 0; i < prevBatteryFaults_.size(); ++i)
+//    {
+//        if (i > 0)
+//            str += "|||||";
+//        str += prevBatteryFaults_[i];
+//    }
+//    qDebug() << str << "\n";
+
+    if (batteryFaults != prevBatteryFaults_ && !batteryFaults.empty() && !std::includes(prevBatteryFaults_.begin(), prevBatteryFaults_.end(), batteryFaults.begin(), batteryFaults.end()))
```

Comment:
Doesn't your "std::includes()" already take care of "batteryFaults != previousBatteryFaults_"?

I know that this way could lead to a faster function, but I think having three conditions decreases readability.

Reply:
Sure, I changed it!